## Data

This data was scraped from woldometers.info on 2022-05-14 by Joseph Assaker.

225 countries are represented in this data.

All of countries have records dating from 2020-2-15 until 2022-05-14 (820 days per country).
That's with the exception of China, which has records dating from 2020-1-22 until 2022-05-14 (844 days per country), and Palau which has records dating from 2021-8-25 until 2022-05-14 (263 days per country)

There are two files in the dataset

1) Summary Data Columns Description:
* **country:** designates the Country in which the the row's data was observed.
* **continent:** designates the Continent of the observed country.
* **total_confirmed:** designates the total number of confirmed cases in the observed country.
* **total_deaths:** designates the total number of confirmed deaths in the observed country.
* **total_recovered:** designates the total number of confirmed recoveries in the observed country.
* **active_cases:** designates the number of active cases in the observed country.
* **serious_or_critical:** designates the estimated number of cases in serious or critical conditions in the observed country.
* **total_cases_per_1m_population:** designates the number of total cases per 1 million population in the observed country.
* **total_deaths_per_1m_population:** designates the number of total deaths per 1 million population in the observed country.
* **total_tests:** designates the number of total tests done in the observed country.
* **total_tests_per_1m_population:** designates the number of total test done per 1 million population in the observed country.
* **population:** designates the population count in the observed country.
    
2) Daily Data Columns Description:
* **date:** designates the date of observation of the row's data in YYYY-MM-DD format.
* **country:** designates the Country in which the the row's data was observed.
* **cumulative_total_cases:** designates the cumulative number of confirmed cases as of the row's date, for the row's country.
* **daily_new_cases:** designates the daily new number of confirmed cases on the row's date, for the row's country.
* **active_cases:** designates the number of active cases (i.e., confirmed cases that still didn't recover nor die) on the row's date, for the row's country.
* **cumulative_total_deaths:** designates the cumulative number of confirmed deaths as of the row's date, for the row's country.
* **daily_new_deaths:** designates the daily new number of confirmed deaths on the row's date, for the row's country.

Source of the Dataset : https://www.kaggle.com/datasets/josephassaker/covid19-global-dataset

## Import

### Import Data and Required libraries

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
#importing daily dataframe
daily_df = pd.read_csv('data/worldometer_coronavirus_daily_data.csv')
daily_df.head()

date      country  cumulative_total_cases  daily_new_cases  \
0  2020-2-15  Afghanistan                     0.0              NaN   
1  2020-2-16  Afghanistan                     0.0              NaN   
2  2020-2-17  Afghanistan                     0.0              NaN   
3  2020-2-18  Afghanistan                     0.0              NaN   
4  2020-2-19  Afghanistan                     0.0              NaN   

   active_cases  cumulative_total_deaths  daily_new_deaths  
0           0.0                      0.0               NaN  
1           0.0                      0.0               NaN  
2           0.0                      0.0               NaN  
3           0.0                      0.0               NaN  
4           0.0                      0.0               NaN

In [6]:
#importing summary Data
summary_df = pd.read_csv('data/worldometer_coronavirus_summary_data.csv')
summary_df.head()

country continent  total_confirmed  total_deaths  total_recovered  \
0  Afghanistan      Asia           179267        7690.0         162202.0   
1      Albania    Europe           275574        3497.0         271826.0   
2      Algeria    Africa           265816        6875.0         178371.0   
3      Andorra    Europe            42156         153.0          41021.0   
4       Angola    Africa            99194        1900.0          97149.0   

   active_cases  serious_or_critical  total_cases_per_1m_population  \
0        9375.0               1124.0                           4420   
1         251.0                  2.0                          95954   
2       80570.0                  6.0                           5865   
3         982.0                 14.0                         543983   
4         145.0                  NaN                           2853   

   total_deaths_per_1m_population  total_tests  total_tests_per_1m_population  \
0                           190.0     951337.0                        23455.0   
1                          1218.0    1817530.0                       632857.0   
2                           152.0     230861.0                         5093.0   
3                          1974.0     249838.0                      3223924.0   
4                            55.0    1499795.0                        43136.0   

   population  
0    40560636  
1     2871945  
2    45325517  
3       77495  
4    34769277

## Data Preparation 

### Basic Analysis

In [8]:
#shape of both data
daily_df.shape

(184787, 7)

We have 184787 number of records and 7 columns in daily dataset

In [9]:
summary_df.shape

(226, 12)

we have 226 number of records and 12 columns in summary data

In [10]:
#check data types
daily_df.dtypes

date                        object
country                     object
cumulative_total_cases     float64
daily_new_cases            float64
active_cases               float64
cumulative_total_deaths    float64
daily_new_deaths           float64
dtype: object

* Here date needed to be fixed, it should be datatime instead of object

In [11]:
#check if there is any null values
daily_df.isna().sum()

date                           0
country                        0
cumulative_total_cases         0
daily_new_cases            10458
active_cases               18040
cumulative_total_deaths     6560
daily_new_deaths           26937
dtype: int64

There are many null values in the data set but wrt to overall data size the  null values are minor.

#### Statistical Summary of Data

In [12]:
daily_df.describe()

cumulative_total_cases  daily_new_cases  active_cases  \
count            1.847870e+05    174329.000000  1.667470e+05   
mean             7.251089e+05      2987.633285  6.239283e+04   
std              3.681471e+06     17803.232663  3.955641e+05   
min              0.000000e+00      -322.000000 -1.432100e+04   
25%              1.099000e+03         0.000000  6.000000e+01   
50%              1.775600e+04        58.000000  1.386000e+03   
75%              2.238085e+05       728.000000  1.462050e+04   
max              8.420947e+07    909610.000000  1.793543e+07   

       cumulative_total_deaths  daily_new_deaths  
count             1.782270e+05     157850.000000  
mean              1.388600e+04         39.831834  
std               6.049521e+04        181.102770  
min               0.000000e+00        -39.000000  
25%               2.400000e+01          0.000000  
50%               3.040000e+02          1.000000  
75%               4.111000e+03         12.000000  
max               1.026646e+06       5093.000000

#### Categorical Summary of Data

In [13]:
daily_df.describe(include = 'O')

date country
count       184787  184787
unique         844     226
top     2021-11-28   China
freq           226     844

* There are total 226 countries about which we're having the data
* China is the most frequent country in the data

In [21]:
#convert data into date time
daily_df['date'] = pd.to_datetime(daily_df['date'])